### PR TITLE
[SIL-55] Login screen

### PR DESCRIPTION
**Pull request by nowakweronika**
### Description
*Design improvement to be consistent with figma.*

### Why?
*For better user experience.*

### Demo
Screenshot 2023-02-28 at 09 23 38

### Documentation
- [SIL-55](https://silentus-au.atlassian.net/browse/SIL-55)
- Link to figma: [Figma](https://www.figma.com/file/yNbcX8VTwufnf7HacYz5Zg/Material-3-Design-Kit-(Community)?node-id=53295%3A27471&t=zAdwPa2cauUTCymx-0)

### CODE REVIEW

File: app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                painter = painterResource(id = R.drawable.ic_github),
+                contentDescription = stringResource(R.string.github_icon),
+                tint = Color.Black,
+            )
+            Text(
+                modifier = Modifier.padding(start = 8.dp, top = 8.dp, bottom = 8.dp),
+                text = stringResource(id = R.string.login),
+                color = Pink40,
+            )
+        }
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt]: **Suggestion:** Extracting this to separate function can be beneficial. 
For example as `LoginButton()`

File: app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        modifier = Modifier.fillMaxSize(),
+        verticalArrangement = Arrangement.SpaceEvenly,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.loudius_logo),
+            contentDescription = stringResource(
+                R.string.login_screen,
+            ),
+        )
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt]: **Suggestion: ** Extracting this to separate method can be beneficial 
For example `LoginImage()`. But it's less beneficial then extracting LoginButton.

File: app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                .padding(horizontal = 46.dp),
+        ) {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_github),
+                contentDescription = stringResource(R.string.github_icon),
+                tint = Color.Black,
+            )
+            Text(
+                modifier = Modifier.padding(start = 8.dp, top = 8.dp, bottom = 8.dp),
+                text = stringResource(id = R.string.login),
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt]: **Question** Are you sure that `log in` should be written with space?

File: app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    context.startActivity(intent)
+}
+
+private fun buildAuthorizationUrl() = BASE_URL + AUTH_PATH + NAME_PARAM_CLIENT_ID + CLIENT_ID
+
+@Preview(showSystemUi = true, showBackground = true)
+@Composable
+fun LoginScreenPreview() {
+    LoginScreen()
+}
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt]: **Issue:** I had a problem with rendering this preview but adding `MaterialTheme()` clause fixed that.

File: app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt (hunk 3 above)
Comment (nowakweronika) [reply app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt]: I think so :)
https://dictionary.cambridge.org/dictionary/english-polish/log-in
https://dictionary.cambridge.org/dictionary/english-polish/login

**Review by Krzysiudan — APPROVED** [review]
Lgtm!

**Pull request merged**